### PR TITLE
Allow PTransforms to be applied directly to dataframes.

### PR DESCRIPTION
**Pull request by robertwb**
This'll be especially handy for applying RunInference.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @damccorm

File: sdks/python/gen_protos.py [hunk 1]
```diff
@@ -123,7 +123,19 @@ def generate_urn_files(out_dir, api_path):
...
+        list,
+        _message.RepeatedScalarContainer,
+        _message.RepeatedCompositeContainer,
+    )  # pylint: disable=c-extension-no-member
+  except ImportError:
+    from google.protobuf.internal import containers
+    list_types = (
+        list,
+        containers.RepeatedScalarFieldContainer,
+        containers.RepeatedCompositeFieldContainer)
```
Comment (damccorm) [inline sdks/python/gen_protos.py:138]: Trying to follow why we need this - is this from having an old protobuf installed? It looks like we previously used `google.protobuf.pyext._message` before https://github.com/apache/beam/commit/5cb1711c39c2ebc3005363f938ca4a4eee774333, is there a reason we're now using `google.protobuf.internal.containers`?

cc/ @AnandInguva @tvalentyn

Comment (AnandInguva) [reply sdks/python/gen_protos.py:138]: I think even with older protobuf, we would need `google.protobuf.pyext._message` when `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION==cpp` but we might need `google.protobuf.internal.containers` when `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python`. 

commented at https://github.com/apache/beam/pull/25914#discussion_r1144970641

File: sdks/python/apache_beam/dataframe/frames.py [hunk 2]
```diff
@@ -5394,6 +5396,24 @@ def func(df, *args, **kwargs):
...
                                        base=pd.DataFrame))
 
+# Allow dataframe | SchemaTransform
+def _create_maybe_elementwise_or(base):
+  elementwise = frame_base._elementwise_method(
+      '__or__', restrictions={'level': None}, base=base)
+
+  def _maybe_elementwise_or(self, right):
+    if isinstance(right, PTransform):
+      return convert.to_dataframe(convert.to_pcollection(self) | right)
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/dataframe/frames.py:5406]: Added a note. Once we push the batching stuff through, there could be little to no overhead here, but that's future work.

File: sdks/python/gen_protos.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/gen_protos.py]: Sorry, this was part of another PR. That's been merged; rebased this PR.

**Review by damccorm — APPROVED** [review]
> Just checking whether "I think this is fine" is an LGTM.

LGTM - I did want to block on the proto changes which is why I didn't give the explicit LGTM, sorry for the ambiguity

**Comment by robertwb** [comment]
No problem. And, yes, the proto changes were merged separately.

**Pull request merged**